src/StreetNameRegistry/Municipality/Municipality_StreetName.cs

```diff
@@ -80,6 +81,11 @@ public void ProposeStreetName(Names streetNameNames, PersistentLocalId persisten
                 throw new StreetNamePersistentLocalIdAlreadyExistsException();
             }
 
+            if (desiredStatus != StreetNameStatus.Proposed && desiredStatus != StreetNameStatus.Current)
+            {
+                throw new StreetNameHasInvalidDesiredStatusException();
+            }
+
```

Comment:
I don't know if this check is really needed, we will lock the municipality before the data is gone out. We don't do a removed check of the previous streetname either. Also very hard to course correct.